app/3.0/docs/devguide/style-shadow-dom.md

```diff
@@ -429,260 +471,273 @@ And you can select by slot name:
+To ensure that your styles behave according to the Shadow DOM v1 specifications in all browsers, use `custom-style` when you define *document-level* styles:
 
-`style.css` { .caption}
-```css
-.red { color: red; }
-.green { color: green; }
-.blue { color: blue; }
-```
+index.html { .caption }
 
-`x-foo.html` { .caption}
-```html
-<!-- Include the styles, and use them in x-foo. -->
-<dom-module id="x-foo">
-  <link rel="import" type="css" href="style.css">
-  <template>
-    <p class="red">I wanna be red</p>
-    <p class="green">I wanna be green</p>
-    <p class="blue">I wanna be blue</p>
-  </template>
-  ...
-</dom-module>
-```
-
-`index.html` { .caption}
 ```html
-<link rel="import" href="x-foo.html">
-
-<x-foo></x-foo>
+<custom-style>
+  <style>
+    /* Document-level styles go here */
+  </style>
+</custom-style>
 ```
 
-## Use `custom-style` in document-level styles {#custom-style}
+`custom-style` is not included in the `@polymer/polymer/polymer-element.js` package. Import `custom-style` from `@polymer/polymer/lib/elements/custom-style.js`:
 
-Browsers that implement the current Shadow DOM v1 specifications will automatically encapsulate styles, scoping them to the elements in which they were defined.
-
-Some browsers have not implemented the Shadow DOM v1 specifications. To make sure your apps and elements display correctly in these browsers, you'll need to use `custom-style` to ensure that styling information doesn't "leak" into the local DOM of your elements.
+index.html { .caption }
 
-`custom-style` enables a set of polyfills that ensure that styles in your apps and elements behave as you would expect from the Shadow DOM v1 specifications, even in browsers that don't implement these specifications.
+```html
+<head>
+  <script src="./custom-element.js" type="module"> 
+  <script src="../@polymer/polymer/lib/elements/custom-style.js" type="module"> 
 
-To ensure that your styles behave according to the Shadow DOM v1 specifications in all browsers, use `custom-style` when you define *document-level* styles. `custom-style` is not included with `Polymer.Element` and must be imported separately.
-`custom-style` is included with the legacy `polymer.html` import.
+  <custom-style>
+    <style>
+      /* Document-level styles go here */
+    </style>
+  </custom-style>
+</head>
+<body>
+  <custom-element></custom-element>
+</body>
+```
 
 *Note: You should only use `custom-style` to define styles for the main document. To define styles for an element's local DOM, just use a `<style>` block.*
 
 ### Examples
 
-In the first code sample, the style for the `p` element “leaks” into Paragraph B in browsers that haven’t implemented the Shadow DOM v1 specs. In the second code sample, the developer has used `custom-style` to wrap the style block, preventing this leak.
+In the following code sample, the document-level style in index.html "leaks" into the shadow DOM of `<custom-element>` in browsers that haven’t implemented the Shadow DOM v1 specs.
```

Comment:
Note for the future: at some point (not right now) we should clarify what we mean by "leaking." It could be confusing, because inheritance is also a form of "leaking," right?

Specifically, what we mean when we say styles can't leak is that _a selector in one scope (i.e., in the main document) can't match an element in a different scope (like in a shadow root)... Except in a couple of very specific  instances, namely :host and ::slotted() ...

Anyway, think on it... We should convey that at some point.

Reply:
agree